vendor/assets/javascripts/checkbox_modal.js

```diff
@@ -0,0 +1,26 @@
+// allow a cancel button to work properly on checkbox modal
+Cmsify.CheckboxModal = function(el) {
+  this.$el = $(el);
+  this.$el.addClass('uk-modal');
+  this.checked = {};
+  this.modal = UIkit.modal(el);
+  this.modal.on('show.uk.modal', function() {
+    this.storeCheckboxes();
+  }.bind(this));
+  this.$el.find('.js-cancel-' + this.$el.attr('id')).each(function(index, cancel) {
+    $(cancel).on('click', this.loadCheckboxes.bind(this)); 
```

Comment:
The problem with submitting from the "Done" button press is if you're on the new collection page and you start adding items before you have filled out the name of the new collection and then hit done you'll lose all the items that you tried to add to the collection after it tries and fails to do a POST.